### PR TITLE
fixes #1010 fix SpringCDIExtension applicationContext and serialization/deserialization of DatawavePrincipals

### LINKED ISSUE #1010: BulkResults not properly initializing CDI injected objects
SpringCDIExtension not properly initializing the ClassPathXmlApplicationContext -- must split the resource string on commas

This was noticed when a running QueryLogic reached a point in the DocumentTransformer where it was adding cardinality information and called Document.getTimestamp() whiched caused a metadata update and a reference to MarkingFunctions which had to be resolved through Weld/CDI/Spring and failed.

CallerPrincipal and ServerPrincipal encode/decode isn't working

Also -- use an Apache Base64 class for encoding/decoding instead of infinispan

### PR DESCRIPTION
**Pull request by billoley**
SpringCDIExtension not properly initializing the ClassPathXmlApplicationContext -- must split the resource string on commas

This was noticed when a running QueryLogic reached a point in the DocumentTransformer where it was adding cardinality information and called Document.getTimestamp() whiched caused a metadata update and a reference to MarkingFunctions which had to be resolved through Weld/CDI/Spring and failed.

CallerPrincipal and ServerPrincipal encode/decode isn't working

Also -- use an Apache Base64 class for encoding/decoding instead of infinispan

### CODE REVIEW

File: web-services/map-reduce-embedded/src/main/java/datawave/security/system/EmbeddedCallerPrincipalProducer.java [hunk 1]
```diff
@@ -7,7 +7,11 @@
 import javax.interceptor.Interceptor;
 
 import datawave.security.authorization.DatawavePrincipal;
-import org.infinispan.commons.util.Base64;
+import org.apache.commons.codec.binary.Base64;
```
Comment (keith-ratcliffe) [inline web-services/map-reduce-embedded/src/main/java/datawave/security/system/EmbeddedCallerPrincipalProducer.java:10]: Does this allow us to remove the infinispan dependency from the POM?

File: web-services/map-reduce-embedded/src/main/java/datawave/security/system/EmbeddedCallerPrincipalProducer.java [hunk 2]
```diff
@@ -31,8 +35,21 @@ public DatawavePrincipal produceCallerPrincipal() {
...
     private void initializeCallerPrincipal() {
         String encodedCallerPrincipal = System.getProperty("caller.principal");
-        if (encodedCallerPrincipal == null)
-            throw new IllegalStateException("System property caller.principal must be set to a serialized, gzip'd, base64 encoded principal.");
-        callerPrincipal = (DatawavePrincipal) Base64.decodeToObject(encodedCallerPrincipal);
+        if (encodedCallerPrincipal == null) {
+            throw new IllegalStateException("System property caller.principal must be set to a serialized, base64 encoded principal.");
+        }
+        ByteArrayInputStream bais = new ByteArrayInputStream(Base64.decodeBase64(encodedCallerPrincipal));
+        try {
```
Comment (keith-ratcliffe) [inline web-services/map-reduce-embedded/src/main/java/datawave/security/system/EmbeddedCallerPrincipalProducer.java:42]: Try-with-resources instead?

Comment (billoley) [reply web-services/map-reduce-embedded/src/main/java/datawave/security/system/EmbeddedCallerPrincipalProducer.java]: Updated to use try with resources

File: web-services/map-reduce-embedded/src/main/java/datawave/security/system/EmbeddedCallerPrincipalProducer.java (hunk 1 above)
Comment (billoley) [reply web-services/map-reduce-embedded/src/main/java/datawave/security/system/EmbeddedCallerPrincipalProducer.java:10]: No - it looks like infinispan is used elsewhere.

Comment (billoley) [reply web-services/map-reduce-embedded/src/main/java/datawave/security/system/EmbeddedCallerPrincipalProducer.java:10]: Looks like we can probably delete it from the map-reduce-embedded pom

**Review by keith-ratcliffe — DISMISSED** [review]

**Review by keith-ratcliffe — APPROVED** [review]

**Pull request merged**